discovery-provider/src/utils/cid_metadata_client.py

```diff
@@ -181,9 +187,126 @@ async def fetch_metadata_from_gateway_endpoints(
 
             except asyncio.TimeoutError:
                 logger.info(
-                    "IPFSCLIENT | fetch_metadata_from_gateway_endpoints TimeoutError"
+                    "CIDMetadataClient | fetch_metadata_from_gateway_endpoints TimeoutError"
                 )
             except Exception as e:
-                logger.info("IPFSCLIENT | Error in fetch cid metadata")
+                logger.info("CIDMetadataClient | Error in fetch cid metadata")
                 raise e
         return cid_metadata
+
+    def fetch_cid_metadata(
```

Comment:
i actually liked this function better in index.py. this now couples cid metadata fetching with POA block processing logic. seems cleaner to have this just be a generic interface where index.py (and future sol equivalent) would just call fetch_metadata_from_gateway_endpoints